CRM/Core/DAO.php

```diff
@@ -1647,17 +1649,13 @@ public static function escapeString($string) {
     if (!$_dao) {
       // If this is an atypical case (e.g. preparing .sql files
       // before Civi has been installed), then we fallback to
-      // DB-less escaping helper (mysql_real_escape_string).
+      // DB-less escaping helper (addslashes). This is unsafe
+      // so should only be used on trusted strings.
       // Note: In typical usage, escapeString() will only
       // check one conditional ("if !$_dao") rather than
       // two conditionals ("if !defined(DSN)")
       if (!defined('CIVICRM_DSN')) {
-        if (function_exists('mysql_real_escape_string')) {
-          return mysql_real_escape_string($string);
-        }
-        else {
-          throw new CRM_Core_Exception("Cannot generate SQL. \"mysql_real_escape_string\" is missing. Have you installed PHP \"mysql\" extension?");
-        }
+        return addslashes($string);
```

Comment:
addslashes isn't the same behaviour as mysql_real_escape_string, IMO it's better to fix this like https://github.com/mollux/civicrm-core/commit/7e5e2fd51e0b08af24bbedbb1a06d127410f560e#diff-6329da386c1213c4300469cf93a3d00cL1645